core/storage/btree.rs

```diff
@@ -3882,41 +3881,51 @@ impl BTreeCursor {
                     payload,
                     first_overflow_page,
                     payload_size,
+                    ..
                 }) => {
                     // TODO: implement efficient comparison of records
                     // e.g. https://github.com/sqlite/sqlite/blob/master/src/vdbeaux.c#L4719
                     return_if_io!(self.read_record_w_possible_overflow(
                         payload,
                         first_overflow_page,
-                        payload_size,
+                        payload_size
                     ));
+
                     let key_values = key.to_index_key_values();
                     let record = self.get_immutable_record();
                     let record = record.as_ref().unwrap();
                     let record_same_number_cols = &record.get_values()[..key_values.len()];
-                    let order = compare_immutable(
+                    compare_immutable(
                         key_values,
                         record_same_number_cols,
                         self.key_sort_order(),
                         &self.collations,
-                    );
-                    match order {
-                        Ordering::Less | Ordering::Equal => {
-                            break;
-                        }
-                        Ordering::Greater => {}
+                    )
+                }
+            };
+
+            match comparison_result {
+                Ordering::Equal => {
+                    result_index = mid;
+                    break;
+                }
+                Ordering::Greater => {
+                    low = mid + 1;
```

Comment:
What if page doesn't have exact matching for the key? I think previous implementation of the method returned first key which is greater than or equal to the search value - but here we can end up with `result_index` equals to `cell_count` (although lower bound is actually in the middle of the page)